### PR TITLE
Delete the YAML linter since it's not part of the normal `conan create`

### LINKED ISSUE #14964: [service] Reduce post push errors from linters
### What is your problem/feature request?

I am experiencing some noisy feedback in https://github.com/conan-io/conan-center-index/pull/14962:

- the linter for yaml fails for totally unrelated changes
- the build keeps running in parallel, so it might be wasting CI resources
- I have used the hooks locally, but didn't realize they are not raising errors, only printing them (I didn't see a red line there)

I think we should reconsider some aspects:
- Do not fail for non-functional requirements in CI. It is nice to try to have some consistency, but failing for an extra blank space that is not affecting functionality, from code that is unrelated, and that is not being checked at all locally
- Try to move some of the checking earlier in the process, try git hooks for example, something that automates it, requiring less configuration steps from users, but better enforcements of some policies
- In any case, if there is doubt, it is better to not error from style and format checks

### PR DESCRIPTION
**Pull request by prince-chrismc**
closes #14964

- delete all the "extras" of the YAML linter that are causing noise

I also fixed a typo :)

### CODE REVIEW

File: docs/adding_packages/conandata_yml_format.md [hunk 1]
```diff
@@ -146,98 +135,3 @@ patches:
       patch_type: "portability"
       patch_source: "https://a-url-to-a-pull-request-mail-list-topic-issue-or-question"
 ```
-
```
Comment (SSE4) [inline docs/adding_packages/conandata_yml_format.md:149]: why do you delete this chunk of the documentation?
it doesn't seem to be related to linter at all

**Comment by SSE4** [comment]
why remote YAML linter altogether?
it could just produce warnings, but avoid to fail the build :thinking:

**Comment by ericLemanissier** [comment]
Strongly opposed to this change. Real errors are caught by this linter

**Comment by prince-chrismc** [comment]
These features are already implemented in the hooks and no one notices those
https://github.com/conan-io/hooks/blob/5bff54624c5c6b01cd5261b377a2f18d53988bd8/hooks/conan-center.py#L50-L51

This yaml lint was adding styling checks and GitHub annotations.

***

I am not in favor of this either but it took a whole 10 minutes to showcase what they were doing and make it more clear to other who've not been following their development.

File: docs/adding_packages/conandata_yml_format.md (hunk 1 above)
Comment (prince-chrismc) [reply docs/adding_packages/conandata_yml_format.md:149]: These are not required and not enforced + the "rules" are duplicate of https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#rules so it seemed appropriate since they were no longer referenced

File: .github/workflows/linter-yaml.yml [hunk 2]
```diff
@@ -1,125 +0,0 @@
...
-
-      - name: Install dependencies
-        if: steps.changed_files.outputs.any_changed == 'true'
-        run: pip install yamllint strictyaml argparse
-
-      - name: Run linter (config.yml)
-        if: steps.changed_files.outputs.any_changed == 'true' && always()
-        run: |
-          echo "::add-matcher::linter/yamllint_matcher.json"
-          yamllint --config-file linter/yamllint_rules.yml -f standard ${{ env.CONFIG_FILES_PATH }}
```
Comment (ericLemanissier) [inline .github/workflows/linter-yaml.yml:39]: AFAIK, yamllint is not run by conan-center hooks, and it catches real errors, which have to prevent PR from being merged, for example in case of key duplication.

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
see also other potential hard errors which should be caught by the linter:
- https://github.com/conan-io/conan-center-index/pull/14912
- https://github.com/conan-io/conan-center-index/pull/14911

I agree that all other errors regarding formatting or comments should be silenced, but please don't throw the baby out with the bathwater

File: .github/workflows/linter-yaml.yml (hunk 2 above)
Comment (SSE4) [reply .github/workflows/linter-yaml.yml:39]: we need to split error for categories (it should be possible, right?)
- style errors like extra space should be plain warnings, they shouldn't fail a job
- errors like duplicated keys should be real errors, blocking further build (and perhaps, aborting build execution)

File: docs/adding_packages/conandata_yml_format.md (hunk 1 above)
Comment (SSE4) [reply docs/adding_packages/conandata_yml_format.md:149]: these are not "rules", it's a documentation/reference on our `conandata.yml` format. which custom fields it has, which values are allowed and so on. it should be left.
in general, "delete linter" PR shouldn't touch unrelated parts of the documentation. in other words, change should be atomic and isolated.

**Comment by SSE4** [comment]
> This yaml lint was adding styling checks and GitHub annotations.
> 

I like the annotations, they are way easier to discover than messages from the hook in megabyte long build log.
IMO style and structure/integrity changes should be separated. style checks could be warnings, or just not displayed at all by default (they could be printed into GH actions log, for instance, but never showed in annotations, to avoid extra noise).

**Comment by prince-chrismc** [comment]
This was not actually remove them, I just wanted to showcase how much they are doing in comparison to what the hooks can do. 
I'll convert to draft so its more clear 😄
